server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -176,17 +177,29 @@ private long calculateQuantity(Subscription sub) {
             if (virtQuantity != null) {
                 Product poolProduct = sub.getProduct();
                 // Favor derived products if they are available
+                // Set the pool's product to the sub's derived product
                 if (sub.getDerivedProduct() != null) {
                     poolProduct = sub.getDerivedProduct();
                 }
 
-                Pool derivedPool = helper.createPool(sub, poolProduct.getId(),
+                Pool newPool = helper.createPool(sub, poolProduct.getId(),
                     virtQuantity, virtAttributes);
 
+                // Set the pool's provided products to the sub's derived provided products
+                Set<ProvidedProduct> providedBySub = new HashSet<ProvidedProduct>();
+                if (sub.getDerivedProduct() != null) {
+                    if (sub.getDerivedProvidedProducts() != null) {
+                        for (Product p : sub.getDerivedProvidedProducts()) {
+                            providedBySub.add(new ProvidedProduct(p.getId(), p.getName(), newPool));
+                        }
+                    }
+                    newPool.setProvidedProducts(providedBySub);
+                }
+
```

Comment:
Looks like we need to add a test then!  The reproducer case is

* deploy with test data
* Run psql -U candlepin candlepin -c "SELECT a.productid, a.id, b.subscriptionid, b.subscriptionsubkey  FROM cp_pool a, cp_pool_source_sub b WHERE a.id = b.pool_id AND a.productid like 'awesomeos-server-basic-%c' ORDER BY a.startdate DESC;"
* Refresh pools
* Run the same query

The results should be identical.  The test data has a lot of extra stuff in it, so I have a stripped down version that narrows it to the affected products.  Holler if you want it.